nat-lab/tests/conftest.py

```diff
@@ -363,3 +364,15 @@ def copy_file_from_container(container_name, src_path, dst_path):
         )
     except subprocess.CalledProcessError:
         print(f"Error copying log file {src_path} from {container_name} to {dst_path}")
+
+
+async def collect_mac_diagnostic_reports():
+    print("Collect mac diagnostic reports")
+    async with mac_vm_util.new_connection(reenable_nat=False) as connection:
+        if isinstance(connection, SshConnection):
```

Comment:
if you were to add `download` method to Connection:ABC and docker_connection, there would be no need to check if this is `SshConnection`. Creating `new_connection` from mac vm util I automatically assume I should get `SshConnection`

Reply:
I like this idea in general, but I think that would make sense in that case to add an OS check in that function. Yes, a mac connection will always be `SshConnection` but an `SshConnection` can also be a windows connection. As long as it's only called with a mac connection it's of course not a functional issue, but I think it would be clearer if there was either a `if is_mac:` check or a comment in the function saying it's only for mac

Reply:
Done